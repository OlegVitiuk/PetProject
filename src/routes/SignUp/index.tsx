import { Button, Form, Icon, Input } from "antd";
import { FormComponentProps } from "antd/lib/form";
import { Link, Redirect } from "react-router-dom";
import * as React from "react";
import { connect } from "react-redux";
import { AppState } from "store";
import styles from "./signUp.module.scss";
import { ThunkDispatch } from "redux-thunk";
import { signUp } from "actions/authActions";
import { bindActionCreators, Action } from "redux";
import { signUpData } from "types/signUp";

interface UserFormProps extends FormComponentProps {}

interface LinkDispatchProps {
  signUp: (userData: signUpData) => void;
}

interface LinkStateProps {
  authError: string;
  isLoggedIn: boolean;
}

interface SignUpProps {
  authError: string;
  isLoggedIn: boolean;
}

type Props = UserFormProps & LinkDispatchProps & SignUpProps;

class SignUp extends React.Component<Props, any> {
  public handleSubmit = (e: React.SyntheticEvent) => {
    const { signUp } = this.props;

    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        console.log("Received values of form: ", values);

        signUp(values);
      }
    });
  };

  public render() {
    const { getFieldDecorator } = this.props.form;
    const { isLoggedIn, authError } = this.props;

    return (
      <React.Fragment>
        {isLoggedIn && <Redirect to="/" />}
        <div className={styles.signUpWrapper}>
          <Form onSubmit={this.handleSubmit} className={styles.signUpForm}>
            {authError && (
              <span className={styles.errorMessage}>{authError}</span>
            )}
            <h1>Sign Up</h1>

            <Form.Item>
              {getFieldDecorator("email", {
                rules: [
                  { required: true, message: "Please input your email!" },
                  {
                    type: "email",
                    message: "The input is not valid email!"
                  }
                ]
              })(
                <Input
                  prefix={
                    <Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />
                  }
                  placeholder="Email"
                />
              )}
            </Form.Item>
            <Form.Item>
              {getFieldDecorator("password", {
                rules: [
                  { required: true, message: "Please input your Password!" }
                ]
              })(
                <Input
                  prefix={
                    <Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />
                  }
                  type="password"
                  placeholder="Password"
                />
              )}
            </Form.Item>
            <Form.Item>
              {getFieldDecorator("firstName", {
                rules: [
                  { required: true, message: "Please input your first name!" }
                ]
              })(
                <Input
                  prefix={
                    <Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />
                  }
                  placeholder="First name"
                />
              )}
            </Form.Item>
            <Form.Item>
              {getFieldDecorator("lastName", {
                rules: [
                  { required: true, message: "Please input your last name!" }
                ]
              })(
                <Input
                  prefix={
                    <Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />
                  }
                  placeholder="Last Name"
                />
              )}
            </Form.Item>
            <Form.Item>
              <Button
                type="primary"
                htmlType="submit"
                className={styles.signUpButton}
              >
                Sign-up!
              </Button>
              Already has a account? <Link to="/sign-in">Login!</Link>
            </Form.Item>
          </Form>
        </div>
      </React.Fragment>
    );
  }
}

export default connect(
  (state: AppState): LinkStateProps => ({
    authError: state.auth.authError,
    isLoggedIn: state.firebase.auth.uid
  }),
  (dispatch: ThunkDispatch<any, any, Action>): LinkDispatchProps => ({
    signUp: bindActionCreators(signUp, dispatch)
  })
)(Form.create<UserFormProps>({ name: "sign-in" })(SignUp));
