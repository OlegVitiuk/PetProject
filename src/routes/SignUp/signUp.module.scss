.signUpForm {
  width: 350px;
  margin-top: 5%;
  text-align: center;
}

.signUpButton {
  width: 100%;
}

.signUpWrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}

.errorMessage {
  color: red;
  font-family: "Inconsolata";
  font-size: 12px;
}
