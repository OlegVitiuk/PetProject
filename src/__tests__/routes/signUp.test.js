import React from 'react';
import { shallow } from 'enzyme';
import SignUp from 'routes/SignUp';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

describe('SignUp route', () => {
  let store = {};

  beforeEach(() => {
    store = mockStore({
      auth: {
        authError: 'asdfasfd'
      },
      firebase: {
        auth: {
          uuid: 'dff'
        }
      }
    });
  });

  it('render SignUp component', () => {
    const wrapper = shallow(<SignUp store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
