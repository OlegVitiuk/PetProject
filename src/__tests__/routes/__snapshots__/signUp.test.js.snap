// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign route render SignIn component 1`] = `ShallowWrapper {}`;

exports[`SignUp route render SignUp component 1`] = `ShallowWrapper {}`;
